rotkehlchen/chain/ethereum/modules/aave/aave.py

```diff
@@ -74,33 +77,60 @@ def __init__(
         self.history_lock = Semaphore()
         self.balances_lock = Semaphore()
         self.counterparties = [CPT_AAVE_V1, CPT_AAVE_V2]
+        self.v3_data_provider = self.ethereum.contracts.contract(AAVE_V3_DATA_PROVIDER)
+        self.v3_reserve_cache: dict[EvmToken, ReserveData] = {}
+
+    def _get_v3_reserve_data(self, reserve_token: 'EvmToken') -> 'ReserveData | None':
+        """Retrieves the reserve data from the v3 data provider"""
+        if (
+            (reserve_data := self.v3_reserve_cache.get(reserve_token)) is None and
+            reserve_token.underlying_tokens is not None and
+            len(reserve_token.underlying_tokens) == 1
+        ):
+            reserve_result = self.v3_data_provider.call(
+                node_inquirer=self.ethereum,
+                method_name='getReserveData',
+                arguments=[reserve_token.underlying_tokens[0].address],
+            )
+            reserve_data = ReserveData(
+                liquidity_rate=FVal(reserve_result[5] / RAY),
+                variable_borrow_rate=FVal(reserve_result[6] / RAY),
+                stable_borrow_rate=FVal(reserve_result[7] / RAY),
+            )
+            self.v3_reserve_cache[reserve_token] = reserve_data
+        return reserve_data
 
     def get_balances(
             self,
             given_defi_balances: 'GIVEN_DEFI_BALANCES',
+            given_eth_balances: 'GIVEN_ETH_BALANCES',
     ) -> dict[ChecksumEvmAddress, AaveBalances]:
         with self.balances_lock:
-            return self._get_balances(given_defi_balances)
+            return self._get_balances(
+                given_defi_balances=given_defi_balances,
+                given_eth_balances=given_eth_balances,
+            )
 
     def _get_balances(
             self,
             given_defi_balances: 'GIVEN_DEFI_BALANCES',
+            given_eth_balances: 'GIVEN_ETH_BALANCES',
     ) -> dict[ChecksumEvmAddress, AaveBalances]:
         """Retrieves the aave balances
 
         Receives the defi balances from zerion as an argument. They can either be directly given
         as the defi balances mapping or as a callable that will retrieve the
         balances mapping when executed.
         """
-        aave_balances = {}
+        aave_balances: defaultdict[ChecksumEvmAddress, AaveBalances] = defaultdict(lambda: AaveBalances(lending={}, borrowing={}))  # noqa: E501
```

Comment:
update the docstring of this function since now it include things from the balances